File: Algorithms/isSubsequence.py
# https://leetcode.com/problems/is-subsequence/
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        for c in s:
            pos = t.find(c)
            if pos==-1:
                return False
            t=t[pos+1:]
        return True    
        

File: Algorithms/isSubsequence.cpp
// https://leetcode.com/problems/is-subsequence/
class Solution {
public:
    bool isSubsequence(string s, string t) {
        decltype(t.size()) pos = -1;
        for(auto &c: s) {
            pos = t.find(c, pos+1);
            if (pos == std::string::npos)
                return false;
        }
        return true;
    }
};

// beats 99.1%
